task/runner.go

```diff
@@ -221,6 +221,22 @@ func (r *runner) getAssetAndOS(assetID string) (*api.Asset, drivers.OSSession, e
 
 func (r *runner) publishTaskResult(ctx context.Context, task data.TaskInfo, output *data.TaskOutput, err error) error {
 	resultCh := make(chan event.PublishResult, 1)
+
+	if err != nil {
+		if strings.Contains(err.Error(), "unexpected EOF") {
+			err = errors.New("file download failed")
+		} else if strings.Contains(err.Error(), "error running ffprobe [] exit status 1") ||
+			strings.Contains(err.Error(), "Could not create stream ID") ||
+			strings.Contains(err.Error(), "502 Bad Gateway") ||
+			(strings.Contains(err.Error(), "EOF") && strings.Contains(err.Error(), "error processing file")) {
+			err = errors.New("error processing file")
+		} else if strings.Contains(err.Error(), "MultipartUpload: upload multipart failed") {
+			err = errors.New("error saving file to internal storage")
```

Comment:
Hmm, just thought that we might wanna call these `internal error`s? WDYT?
```suggestion
			err = errors.New("internal error processing file")
		} else if strings.Contains(err.Error(), "MultipartUpload: upload multipart failed") {
			err = errors.New("internal error saving file to storage")
```

Reply:
Yeah, looks better! I was actually thinking about having a `error:` prefix like `error: Internal Error saving file to storage` for more readability, wdyt? For now I pushed your proposal

Reply:
Hmm, I think in the context where these messages will show up, it will already be clear that they correspond to an error message. So we don't need that prefix specifier. e.g.
```json
        "status": {
            "phase": "failed",
            "updatedAt": 1655333678345,
            "errorMessage": "error probing file: error running ffprobe [] exit status 1"
        },
```

Which with the current changes, should become
```json
        "status": {
            "phase": "failed",
            "updatedAt": 1655333678345,
            "errorMessage": "internal error processing file"
        },
```

WDYT? Still think that an `error:` prefix would be necessary?

Reply:
Definitely not necessary, just thought that uppercase errors in texts like `Internal Error` would provide more readability, but we can keep it like it is now! :)

Reply:
Oh I thought you were only proposing an `error:` prefix to the messages. Did not understand the casing suggestion.

I like making these cased like sentences (first world Upper) instead of go errors (all lower) as well! Feel free to update these and just ignore the go vet warnings, it doesn't know what it's doing 😆

An alternative could also be to return a string from this function instead of an error. But you then would have to change the calling function as well, don't bother about it.